modules/search/instant-search/components/scroll-button.jsx

```diff
@@ -0,0 +1,47 @@
+/** @jsx h */
+
+/**
+ * External dependencies
+ */
+import { h, Component } from 'preact';
+import { __ } from '@wordpress/i18n';
+// NOTE: We only import the debounce package here for to reduced bundle size.
+//       Do not import the entire lodash library!
+// eslint-disable-next-line lodash/import-scope
+import debounce from 'lodash/debounce';
+
+class ScrollButton extends Component {
+	componentDidMount() {
+		this.props.enableAutoScrolling && document.addEventListener( 'scroll', this.checkScroll );
+	}
+	componentDidUnmount() {
+		document.removeEventListener( 'scroll', this.checkScroll );
+	}
+
+	checkScroll = debounce( () => {
+		if (
+			this.props.enableAutoScrolling &&
+			window.innerHeight + window.scrollY === document.body.offsetHeight
+		) {
+			this.props.onClick();
+		}
+	}, 100 );
```

Comment:
Why do we debounce for this case? I had the same question for the filtering. Doesn't debouncing here just slow down how responsive the UI feels?

Reply:
We're debouncing because `checkScroll` reads many DOM properties that cause [layout thrashing](https://kellegous.com/j/2013/01/26/layout-performance/). 

When a function fired in rapid succession (as we do with a `scroll` event handler here), it significantly impacts a browser's render performance. This performance degradation is especially noticeable for mobile browsers.

Reply:
Awesome, thanks for the link!